kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java

```diff
@@ -295,6 +313,77 @@ private void decryptAndConvertPasswordProperties(Map<String, Object> modifiedPro
         }
     }
 
+    private void decryptAndConvertCertificatesProperties(Map<String, Object> modifiedProps) throws KuraException {
+
+        for (Entry<String, Object> prop : modifiedProps.entrySet()) {
+            if (prop.getKey().contains("802-1x.keystore.pid")) {
+                String keystorePid = (String) prop.getValue();
+
+                String interfaceName = prop.getKey().split("\\.")[2];
```

Comment:
🤔  I know that it shouldn't happen but... what about VLAN interfaces? They have a `.` in their name.

Reply:
Wouldn't it be better (and more performant) to iterate over the known interfaces and then check if the `modifiedProps` map contains the "[interface-name].802-1x.keystore.pid" key? If you do it this way you also avoid the issue with the VLAN interface names. LMKWYT